lex/lexer.go

```diff
@@ -2623,7 +2552,7 @@ func LexExpression(l *Lexer) StateFn {
 		}
 	}
 	// ensure we don't get into a recursive death spiral here?
-	if len(l.stack) < 1000 {
+	if len(l.stack) < maxDepth || l.overrideDepthLimit {
 		l.Push("LexExpression-clauseStatex", l.clauseState())
 	} else {
 		u.LogThrottle(u.WARN, 10, "Gracefully refusing to add more LexExpression: %s", l.input)
```

Comment:
just noting, we probably want to address this separately, but this always drove me crazy ha. if you blow the stack it just "gracefully" refuses to add expressions, and you get random undefined behavior that (if you're lucky) gets logged, instead of an error. think this sort of thing happens elsewhere too...

Reply:
yeah feels like we should just return error saying too much nested expression, as the parsing itself is going to give you false result vs what was provided as expression

Reply:
yeah i agree :)